okcupyd/xpath.py

```diff
@@ -72,7 +72,13 @@ def apply_(self, tree):
         return tree.xpath(self.xpath)
 
     def one_(self, tree):
-        return self.apply_(tree)[0]
+        try:
```

Comment:
XPathBuilder is a class for generating generic xpath strings that can be supplied to lxml objects. I'm not sure that I think it is appropriate for it to know anything about profiles, or really, anything about okcupyd, for that matter. This check really ought to live in profile.py in the profile_tree method.

One of the problems with throwing an exception at all in the generation of profiles is that a lot of profile attributes are generated lazily (using cached_properties), including profile tree itself, which means that you can't merely try...except the creation of the profile (unless we were to force eager evaluation, which seems undesireable). Maybe the right approach is to have a validate_existence method that either throws an exception or returns false when the profile object points to a non existing profile...? Alternatively, we could slap a decoractor on all of the cached_profiles that ensure the existence of the profile before attempting to retrieve a value to avoid cryptic xpath errors/IndexErrors.